### PR TITLE
Autotool'ify nxdialog subfolder

### PR DESCRIPTION
**Pull request by sunweaver**

### CODE REVIEW

**Review by uli42 — CHANGES_REQUESTED** [review]
please add this patch to allow for hexdecimal window ids (as reported by xwininfo):
```
diff --git a/nxdialog/nxdialog b/nxdialog/nxdialog
index e7ca10aad..379ce0024 100755
--- a/nxdialog/nxdialog
+++ b/nxdialog/nxdialog
@@ -45,8 +45,9 @@
 # If an "NX_CLIENT" environment variable is not provided to nxagent
 # nxcomp library assumes this script is located in /usr/NX/bin/nxclient
 #
-# Example:
+# Examples:
 # nxdialog --dialog yesno --message "message text" --caption "message title" --parent 0
+# nxdialog --dialog pulldown --message "message text" --caption "message title" --window 0x123456 --parent 0
 
 from __future__ import print_function
 
@@ -110,7 +111,7 @@ class PullDownMenu(object):
         """ Shows popup and returns result. """
 
         display = Gdk.Display.get_default()
-        win = GdkX11.X11Window.foreign_new_for_display(display, self.window_id)
+        win = GdkX11.X11Window.foreign_new_for_display(display, int(self.window_id, 0))
 
         menu = Gtk.Menu()
         menu.connect("deactivate", self.menu_deactivate)
@@ -312,7 +313,7 @@ class NxDialogProgram(object):
                             shown")
         parser.add_argument("--parent", type=int, dest="agentpid",
                             help="pid of the nxagent")
-        parser.add_argument("--window", type=int, dest="window",
+        parser.add_argument("--window", dest="window",
                             help="id of window where to embed the \
                             pulldown dialog type")
         # -class, -local, -allowmultiple are unused in nxlibs 3.5.99.18
```

File: nxdialog/Makefile.am [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    $(srcdir)/build-aux/*		\
+    $(srcdir)/Makefile.in		\
+    $(srcdir)/man/Makefile.in		\
+    $(srcdir)/src/Makefile.in		\
+    $(srcdir)/aclocal.m4		\
+    $(srcdir)/config.h.in		\
+    $(srcdir)/config.h.in~		\
+    $(srcdir)/config.guess		\
+    $(srcdir)/config.sub		\
+    $(srcdir)/configure			\
```
Comment (uli42) [inline nxdialog/Makefile.am:18]: there's a tab/space bug in the config.sub line

File: nxdialog/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# NX Dialog Helper Tool
+
+The ``nxdialog`` script can be execute with Python 2.6.x (or higher) and
```
Comment (uli42) [inline nxdialog/README.md]: typo: executeD

File: nxdialog/man/nxdialog.1 [hunk 3]
```diff
@@ -0,0 +1,72 @@
+'\" -*- coding: utf-8 -*-
+.if \n(.g .ds T< \\FC
+.if \n(.g .ds T> \\F[\n[.fam]]
+.de URL
+\\$2 \(la\\$1\(ra\\$3
+..
+.if \n(.g .mso www.tmac
+.TH nxdialog 1 "Jan 2019" "Version 3.5.99.18" "NX Dialog"
```
Comment (uli42) [inline nxdialog/man/nxdialog.1:8]: Hmm, that date looks strange..

File: nxdialog/man/nxdialog.1 [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+         [--window WINDOW] [--class DLGCLASS] [--local]
+         [--allowmultiple]\fR
+
+.SH DESCRIPTION
+\fBnxdialog\fR is a GUI helper that can be used with nxagent to provide
+user interaction on certain events inside nxagent. It adds graphical
+widgets (menus, dialog boxes, etc.) to nxagent that allow a smarter user
+interaction with the agent.
+.PP
+If your \fBnxagent\fR is not using \fBnxdialog\fR by default, then set
```
Comment (uli42) [inline nxdialog/man/nxdialog.1]: Rephrase: nxagent is not using nxdialog by default, set NX_CLIENT to make the nxdialog path known to nxagent.

File: nxdialog/man/nxdialog.1 [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+.TP 8
+.B --local
+Specify that proxy mode is used [currently unimplemented].
+.TP 8
+.B --allowmultiple
+Allow launching more than one dialog with the same message [currently unimplemented].
+
+.SH AUTHOR
+The \fBnxdialog\fR application has originally been derived from Google's
+NeatX client application.
```
Comment (uli42) [inline nxdialog/man/nxdialog.1]: please add my name

**Review by uli42 — APPROVED** [review]
apart from the few comments I made this is ok to getting merged

**Comment by sunweaver** [comment]
> please add this patch to allow for hexdecimal window ids (as reported by xwininfo):
> 
> ```
> diff --git a/nxdialog/nxdialog b/nxdialog/nxdialog
> index e7ca10aad..379ce0024 100755
> --- a/nxdialog/nxdialog
> +++ b/nxdialog/nxdialog
> @@ -45,8 +45,9 @@
>  # If an "NX_CLIENT" environment variable is not provided to nxagent
>  # nxcomp library assumes this script is located in /usr/NX/bin/nxclient
>  #
> -# Example:
> +# Examples:
>  # nxdialog --dialog yesno --message "message text" --caption "message title" --parent 0
> +# nxdialog --dialog pulldown --message "message text" --caption "message title" --window 0x123456 --parent 0
>  
>  from __future__ import print_function
>  
> @@ -110,7 +111,7 @@ class PullDownMenu(object):
>          """ Shows popup and returns result. """
>  
>          display = Gdk.Display.get_default()
> -        win = GdkX11.X11Window.foreign_new_for_display(display, self.window_id)
> +        win = GdkX11.X11Window.foreign_new_for_display(display, int(self.window_id, 0))
>  
>          menu = Gtk.Menu()
>          menu.connect("deactivate", self.menu_deactivate)
> @@ -312,7 +313,7 @@ class NxDialogProgram(object):
>                              shown")
>          parser.add_argument("--parent", type=int, dest="agentpid",
>                              help="pid of the nxagent")
> -        parser.add_argument("--window", type=int, dest="window",
> +        parser.add_argument("--window", dest="window",
>                              help="id of window where to embed the \
>                              pulldown dialog type")
>          # -class, -local, -allowmultiple are unused in nxlibs 3.5.99.18
> ```

Will add this commit after merging the PR.

File: nxdialog/Makefile.am (hunk 1 above)
Comment (sunweaver) [reply nxdialog/Makefile.am:18]: that tab/space flaw is only in the diff, not in the file.

File: nxdialog/README.md (hunk 2 above)
Comment (sunweaver) [reply nxdialog/README.md]: got that, nice catch

File: nxdialog/man/nxdialog.1 (hunk 3 above)
Comment (sunweaver) [reply nxdialog/man/nxdialog.1:8]: > Hmm, that date looks strange..

date bump comes with release, the date is as in all the other man pages

File: nxdialog/man/nxdialog.1 (hunk 4 above)
Comment (sunweaver) [reply nxdialog/man/nxdialog.1]: > Rephrase: nxagent is not using nxdialog by default, set NX_CLIENT to make the nxdialog path known to nxagent.

got that

File: nxdialog/man/nxdialog.1 (hunk 5 above)
Comment (sunweaver) [reply nxdialog/man/nxdialog.1]: > please add my name

done.

**Pull request merged**